packages/aws-rfdk/lib/core/lambdas/nodejs/version-provider/handler.ts

```diff
@@ -0,0 +1,357 @@
+  readonly PatchVersion: string;
+
+  /**
+   * The URLs to installers
+   */
+  readonly Uris: IUris;
+}
+
+export class VersionProvider extends SimpleCustomResource {
+  readonly indexFilePath: string|undefined;
+
+  constructor(indexFilePath?: string) {
+    super();
+    this.indexFilePath = indexFilePath;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  /* istanbul ignore next */ // @ts-ignore
+  public validateInput(data: object): boolean {
+    return this.implementsIVersionProviderProperties(data);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  // @ts-ignore  -- we do not use the physicalId
+  public async doCreate(physicalId: string, resourceProperties: IVersionProviderProperties): Promise<Map<Platform, IVersionedUris>> {
+    /* istanbul ignore next */
+    const indexJson = this.indexFilePath ? this.readInstallersIndex() : await this.downloadInstallerIndex();
+
+    const productSection = indexJson[resourceProperties.product];
+
+    if (!productSection) {
+      throw new Error(`Information about product ${resourceProperties.product} can't be found`);
+    }
+
+    let installers = new Map();
+    if (resourceProperties.platform) {
+      const versionedUris = await this.getUrisForPlatform(
+        resourceProperties.product,
+        productSection,
+        resourceProperties.platform,
+        resourceProperties.versionString);
+
+      if (versionedUris) {
+        installers.set(resourceProperties.platform, versionedUris);
+      }
+    } else {
+      Object.values(Platform).forEach(async p => {
+        const versionedUris = await this.getUrisForPlatform(
+          resourceProperties.product,
+          productSection,
+          p,
+          resourceProperties.versionString);
+
+        if (versionedUris) {
+          installers.set(p, versionedUris);
+        }
+      });
+    }
+
+    return installers;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  /* istanbul ignore next */ // @ts-ignore
+  public async doDelete(physicalId: string, resourceProperties: IVersionProviderProperties): Promise<void> {
+    // Nothing to do -- we don't modify anything.
+    return;
+  }
+
+  private implementsIVersionProviderProperties(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+
+    // TODO: Ignore case
+    if (!value.product || !Object.values(Product).includes(value.product)) {
+      return false;
+    }
+
+    if (value.versionString) {
+      if (null === this.parseVersionString(value.versionString))  { return false; }
+    }
+
+    if (value.platform) {
+      // TODO: Ignore case
+      if (!Object.values(Platform).includes(value.platform))  { return false; }
+    }
+
+    return true;
+  }
+
+  /* istanbul ignore next */ // @ts-ignore
+  private async downloadInstallerIndex() {
+    const productionInfoURL = 'https://downloads.thinkboxsoftware.com/version_info.json';
+
+    // eslint-disable-next-line @typescript-eslint/no-require-imports
+    const parsedUrl = require('url').parse(productionInfoURL);
```

Comment:
do we have to use require imports here? or can we just do a normal import at the top of the file.